wolfcrypt/src/random.c

```diff
@@ -800,6 +809,776 @@ int wc_RNG_TestSeed(const byte* seed, word32 seedSz)
 #endif /* HAVE_HASHDRBG */
 /* End NIST DRBG Code */
 
+#ifdef HAVE_ENTROPY_MEMUSE
+/* Define ENTROPY_MEMUSE_THREAD to force use of counter in a new thread.
+ * Only do this when high resolution timer not otherwise available.
+ */
+
+/* Number of bytes that will hold the maximum entropy bits. */
+#define MAX_ENTROPY_BYTES    (MAX_ENTROPY_BITS / 8)
+/* Number of bits stored for one sample. */
+#define ENTROPY_BITS_USED    8
+
+/* Minimum entropy from a sample. */
+#define ENTROPY_MIN          1
+/* Number of extra samples to ensure full entropy. */
+#define ENTROPY_EXTRA        64
+/* Maximum number of bytes to sample to produce max entropy. */
+#define MAX_NOISE_CNT        (MAX_ENTROPY_BITS * 8 + ENTROPY_EXTRA)
+
+/* Jitter entropy global state initialized. */
+static int entropy_memuse_initialized = 0;
```

Comment:
Should we protect the `entropy_memuse_initialized`  variable (and all the other global flags for this new feature)with Thread Local Storage (TLS) or THREAD_LS_T?

Reply:
No.
Entropy_Init called by wolfCrypt_Init().
Entropy_Final called by wolfCrypt_Cleanup().